garden_ai/backend_client.py

```diff
@@ -59,11 +59,11 @@ def _delete(self, resource: str, payload: dict) -> dict:
     def _get(self, resource: str, params: dict | None = None) -> dict:
         return self._call(requests.get, resource, None, params=params)
 
-    def get_garden(self, doi: str) -> Garden:
+    def get_garden(self, doi: str, client=None) -> Garden:
```

Comment:
this smells a little odd, iirc the only reason `_from_nested_metadata` even accepts a client was a kludge for unit tests. since the GET request to the backend only needs a doi, I think the BackendClient.get_garden should also only take a doi